### ISSUE TITLE
��� in results

### ISSUE THREAD

**Issue by yas375** [SlatherOrg/slather#199]
1. `git clone https://github.com/mattt/FormatterKit.git`
2. `cd FormatterKit`
3. `git checkout abb11c3`
4. `bundle install`
5. `rake spec`
6. `rake coverage:consolidate`
7. `bundle exec slather coverage --html`
8. `open html/TTTTimeIntervalFormatter.m.html`

Result: generated html report shows � instead of some utf characters.

<img width="925" alt="2016-05-01_0206" src="https://cloud.githubusercontent.com/assets/217368/14940990/6fe927e2-0f41-11e6-891b-7d3fca9e27c3.png">
Same in the [report sent to coveralls.io](https://coveralls.io/builds/6001834/source?filename=FormatterKit%2FTTTTimeIntervalFormatter.m#L493).

**Comment by ksuther** [comment]
The problem appears to be here: https://github.com/SlatherOrg/slather/blob/master/lib/slather/project.rb#L200

Changing `'UTF-8', 'binary'` to just `'UTF-8'` fixes the problem, but I'm not sure why it was trying to convert from binary in the first place. `unsafe_profdata_llvm_cov_output` appears to return a UTF-8 string in the first place, at least for me.

**Comment by yas375** [comment]
@ksuther you are right! Removing `binary` from the list of arguments solves the problem to me.
Also replacing `binary` with `UTF-8` does too.

```ruby
unsafe_profdata_llvm_cov_output(binary_path).encode!('UTF-8', :invalid => :replace, undef: :replace)
```
```ruby
unsafe_profdata_llvm_cov_output(binary_path).encode!('UTF-8', 'UTF-8', :invalid => :replace, undef: :replace)
```

I did find the first commit where it was added: https://github.com/SlatherOrg/slather/commit/ad9b9eb7e8dcc9f8ccb7657bf3249f4a4d411e7d
And [here is a related discussion](https://github.com/SlatherOrg/slather/pull/92#issuecomment-164962700) on the topic.

I'm not sure what should be a solution here, but I'm happy to help with verifying any possible fixes :)

**Comment by neonichu** [comment]
We could try encoding as UTF-8 instead and if that throws an encoding error use binary. This should solve this issue and still retain the fallback behaviour that was originally intended.

**Comment by ksuther** [comment]
Yeah, I think that trying this first would work in most cases:

```
unsafe_profdata_llvm_cov_output(binary_path).encode!('UTF-8', :invalid => :replace, undef: :replace)
```

Then fall back to the existing line if that fails. The output of unsafe_profdata_llvm_cov_output is UTF-8 already, at least for me. Looks like that's not always the case, but the fallback would catch that.

**Comment by ksuther** [comment]
Fixed by #259.
